src/blockdev.rs

```diff
@@ -0,0 +1,208 @@
+use std::collections::HashMap;
```

Comment:
Not a blocker here but maybe we should try to sync this code with https://github.com/containers/bootc/blob/main/lib/src/blockdev.rs ?

I could try to refactor that into a separate internal crate that we sync back into bootupd and coreos-installer?

Or of course we could even try to publish it.

Reply:
> Not a blocker here but maybe we should try to sync this code with https://github.com/containers/bootc/blob/main/lib/src/blockdev.rs ?
> 
> I could try to refactor that into a separate internal crate that we sync back into bootupd and coreos-installer?
> 
> Or of course we could even try to publish it.

Thanks, that would be better and make this easier. 
I tried to copy the `bootc/blockdev.rs`, but seems there are a lot of dependencies that also need to sync, for example `Task` and `bootc_utils::CommandRunExt`, so I give up.

Reply:
The idea of https://github.com/coreos/bootupd/pull/820#issuecomment-2603461521 is that the Task dependency is gone. However yes, we'd need to pull in `bootc-utils` too. We could even do that now.

OK I did just that in https://github.com/coreos/bootupd/pull/821

Reply:
Cool, thanks!

Reply:
Can we import https://github.com/containers/bootc/blob/main/lib/src/blockdev.rs like what you did in https://github.com/coreos/bootupd/pull/821 ?

Reply:
https://github.com/containers/bootc/pull/1050